azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/device_client.py

```diff
@@ -4,7 +4,30 @@
 # --------------------------------------------------------------------------------------------
 
 from .internal_client import InternalClient
+import logging
+from threading import Event
+
+logger = logging.getLogger(__name__)
 
 
 class DeviceClient(InternalClient):
-    pass
+    def __init__(self, auth_provider, transport):
+        InternalClient.__init__(self, auth_provider, transport)
+        self._transport.on_transport_c2d_message_received = self._handle_c2d_message_received
+        self.on_c2d_message = None
+
+    def enable_c2d_messages(self, handler_for_c2d=None):
+        enable_complete = Event()
+
+        def callback():
+            enable_complete.set()
+
+        self._transport.enable_c2d_messages(callback, handler_for_c2d)
```

Comment:
>handler_for_c2d [](start = 54, length = 15)

I think this should parallel the way input messages work.  In other words, don't pass the handler to the enable function.  Instead, set the handler as an attribute.